homeassistant/components/ozw/climate.py

```diff
@@ -0,0 +1,356 @@
+        """
+        if len(self._current_mode_setpoint_values) == 1:
+            setpoint = self._current_mode_setpoint_values[0]
+            target_temp = kwargs.get(ATTR_TEMPERATURE)
+            if setpoint is not None and target_temp is not None:
+                setpoint.send_value(target_temp)
+        elif len(self._current_mode_setpoint_values) == 2:
+            (setpoint_low, setpoint_high) = self._current_mode_setpoint_values
+            target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+            if setpoint_low is not None and target_temp_low is not None:
+                setpoint_low.send_value(target_temp_low)
+            if setpoint_high is not None and target_temp_high is not None:
+                setpoint_high.send_value(target_temp_high)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        if not self.values.fan_mode:
+            return
+        # get id for this fan_mode
+        fan_mode_value = _get_list_id(self.values.fan_mode.value[VALUE_LIST], fan_mode)
+        if fan_mode_value is None:
+            _LOGGER.warning("Received an invalid fan mode: %s", fan_mode)
+            return
+        self.values.fan_mode.send_value(fan_mode_value)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        if not self.values.mode:
+            return
+        hvac_mode_value = HVAC_MODE_ZW_MAPPINGS.get(hvac_mode)
+        if hvac_mode_value is None:
+            _LOGGER.warning("Received an invalid hvac mode: %s", hvac_mode)
+            return
+        self.values.mode.send_value(hvac_mode_value)
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new target preset mode."""
+        if not self.values.mode:
+            return
+        if preset_mode == PRESET_NONE:
+            # try to restore to the (translated) main hvac mode
+            await self.async_set_hvac_mode(self.hvac_mode)
+            return
+        preset_mode_value = _get_list_id(
+            self.values.mode.value[VALUE_LIST], preset_mode
+        )
+        if preset_mode_value is None:
+            _LOGGER.warning("Received an invalid preset mode: %s", preset_mode)
+            return
+        self.values.mode.send_value(preset_mode_value)
+
+    @property
+    def device_state_attributes(self):
+        """Return the optional state attributes."""
+        data = super().device_state_attributes
+        if self.values.fan_action:
+            data[ATTR_FAN_ACTION] = self.values.fan_action.value
+        if self.values.valve_position:
+            data[
+                ATTR_VALVE_POSITION
+            ] = f"{self.values.valve_position.value} {self.values.valve_position.units}"
+        return data
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        support = SUPPORT_TARGET_TEMPERATURE
```

Comment:
I think we should only support `SUPPORT_TARGET_TEMPERATURE` if we use a single set point. If we use two set points we should use `SUPPORT_TARGET_TEMPERATURE_RANGE`.

Reply:
```suggestion
        support = 0
        if len(self._current_mode_setpoint_values) == 1:
            support |= SUPPORT_TARGET_TEMPERATURE
```